### ISSUE TITLE
test(twap): revisit err tolerance in geometric twap with high scaling factor stableswap pool 

### ISSUE THREAD

**Issue by p0mvn** [osmosis-labs/osmosis#3770]
Can we make an issue to revisit this error tolerance in the lens of stableswap with high scaling factors?

_Originally posted by @ValarDragon in https://github.com/osmosis-labs/osmosis/pull/3726#discussion_r1050991112_

**Comment by p0mvn** [comment]
Hi @ValarDragon .

For a final ACK for v14 release, can you please confirm if there is nothing else to check here?

I've looked into this, and it seems to me that these concerns of yours related to stableswap pools are not a problem.

The spot price keeper call should handle the case if the result returned from stableswap is not positive:
https://github.com/osmosis-labs/osmosis/blob/cf88b689290eb1a94e3d62497c0a8806e872e320/x/gamm/keeper/pool_service.go#L51

If that happens, an error is returned to twap. Then, twap would set [the spot prices to zero](https://github.com/osmosis-labs/osmosis/blob/cf88b689290eb1a94e3d62497c0a8806e872e320/x/twap/logic.go#L58-L62) and update `LastErrorTime`

Consecutively, if that is the case, there is geometric twap logic that should cover such behavior:
 - If spot price is zero, do not take logarithm of price and return the record with no updates: https://github.com/osmosis-labs/osmosis/blob/cf88b689290eb1a94e3d62497c0a8806e872e320/x/twap/logic.go#L197-L203
 - If there is a problematic stableswap poo that always errors in spot price, `computeTwap` geometric logic should also account for that: https://github.com/osmosis-labs/osmosis/blob/cf88b689290eb1a94e3d62497c0a8806e872e320/x/twap/strategy.go#L46-L48

TLDR;  it looks to me that there is nothing to be done here but would like a final ACK before closing

**Comment by ValarDragon** [comment]
Agreed, thanks for the thorough check!

I think there is still a present concern, independent of TWAP, of maybe we aren't giving enough precision for stableswap due to the precision factors.
